### PR TITLE
Create temporary tables in mysql without key constraints

### LINKED ISSUE #4979: Do not copy constraints from the target table when creating a temporary table in Mysql before inserting
Temporary tables without constraints can be inserted faster,  and there is no need to copy constraints such as the primary key from the target table.

Related thread in Slack: https://prestosql.slack.com/archives/CFLB9AMBN/p1598339154087800

Use `CREATE TABLE AS SELECT WHERE 0=1` instead of `CREATE TABLE LIKE` to copy a table in MySQL.

### PR DESCRIPTION
**Pull request by ssquan**
Fixes #4979

### CODE REVIEW

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 1]
```diff
@@ -307,8 +307,9 @@ public void renameColumn(JdbcIdentity identity, JdbcTableHandle handle, JdbcColu
     @Override
     protected void copyTableSchema(Connection connection, String catalogName, String schemaName, String tableName, String newTableName, List<String> columnNames)
     {
+        // Create a temporary table not only include columns to be inserted but all columns,  Need them for `NOT NULL` check.
         String sql = format(
-                "CREATE TABLE %s LIKE %s",
+                "CREATE TABLE %s AS SELECT * FROM %s WHERE 0 = 1",
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:312]: If we change to CTAS from the current implementation, it breaks the existing behavior when GTID mode is on. We need to look for other approach (e.g. Use CREATE TABLE LIKE only when GTID mode is on or just adding cofiguration).

**Review by ebyhr — CHANGES_REQUESTED** [review]

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java (hunk 1 above)
Comment (findepi) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:312]: do we have tests for this? or not possible? (sorry, dont remember)
(sadly, all checks paassd... )

Comment (ssquan) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:312]: I see, `CTAS` not working in GTID mode. I found this issue  #2251

Comment (ssquan) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:312]: > do we have tests for this? or not possible? (sorry, dont remember)
> (sadly, all checks paassd... )

@ebyhr just file an issue to add tests in GTID mode :) #4997

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 2]
```diff
@@ -268,6 +268,21 @@ else if (varcharType.getBoundedLength() <= 16777215) {
         return super.toWriteMapping(session, type);
     }
 
+    private boolean isGTIDMode(Connection connection)
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:271]: Move this method to after `jsonColumnMapping` method. 
```suggestion
    private boolean isGtidMode(Connection connection)
```

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 3]
```diff
@@ -268,6 +268,21 @@ else if (varcharType.getBoundedLength() <= 16777215) {
...
+    {
+        try (java.sql.Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery("show variables like 'gtid_mode'");
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:274]: We can make this line try-with-resources and please upper case for keywords. 
```suggestion
            ResultSet resultSet = statement.executeQuery("SHOW VARIABLES LIKE 'gtid_mode'");
```

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 4]
```diff
@@ -268,6 +268,21 @@ else if (varcharType.getBoundedLength() <= 16777215) {
...
+            if (resultSet.next()) {
+                return resultSet.getString("Value").equalsIgnoreCase("ON");
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:276]: `ON_PERMISSIVE` is also possible value.

**Review by ebyhr — COMMENTED** [review]
Let me review again after https://github.com/prestosql/presto/pull/5020 is merged. The commit message should be updated now because we changed the code from the initial commit.

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 5]
```diff
@@ -344,6 +348,22 @@ private ColumnMapping jsonColumnMapping()
                 DISABLE_PUSHDOWN);
     }
 
+    private boolean isGtidMode(Connection connection)
+    {
+        try (java.sql.Statement statement = connection.createStatement()) {
+            try (ResultSet resultSet = statement.executeQuery("SHOW VARIABLES LIKE 'gtid_mode'")) {
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:354]: You don't need to split into 2 `try`.

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java (hunk 4 above)
Comment (ssquan) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java]: I use a `not OFF mode` check here to consider `OFF_PERMISSIVE `、`ON__PERMISSIVE` and `ON` modes at the same time. Mysql may use GTID-based replication in `OFF_PERMISSIVE` mode according to [MySQL dev doc](https://dev.mysql.com/doc/refman/5.7/en/replication-options-gtids.html)

**Review by ebyhr — APPROVED** [review]
Please rebase on upstream and modify the commit message. e.g. 
```
Do not copy constraint for temporary table on non-GTID MySQL
```

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**